### PR TITLE
Fixed unexpected behaviour on performing multiple operations (update and skip) in configuration component one after the other (#85zrnt2um)

### LINKED ISSUE #382: Perform multiple operations (update and skip) on configuration component one after the other results in unexpected behaviour
## Current behavior

When skipping a job after updating the job frequency, updates the runTime for the job by taking into consideration the previous frequency of the job and not the updated one.

## Expected behavior

When skipping a job, always the latest job frequency should be considered.

## Steps to reproduce the issue

1. Go to pipeline page
2. Click on a job
3. Update the frequency for the job
4. On successful, skip the same job again
5. See that the job has been skipped by using the previous job frequency and not the latest one.

## Can you handle fixing this bug by yourself?

- [x] YES
- [ ] NO

## Environment details

- Browser: 
- OS: 
- Code Version:

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #382 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/store/modules/job/actions.ts [hunk 1]
```diff
@@ -470,7 +470,7 @@ const actions: ActionTree<JobState, RootState> = {
             'systemJobEnumId_op': 'equals'
           }
         })
-        const fetchJobsResponse = jobs.find((job: any) => job?.jobId === payload.jobId);
+        const fetchJobsResponse = jobs.find((job: any) => job?.data.docs[0].jobId === payload.jobId);
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:473]: Improve fetchJobsResponse variable name and as we have only one response

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts (hunk 1 above)
Comment (k2maan) [reply src/store/modules/job/actions.ts]: Improved the variable name to `jobResponse`. 

As we are fetching both draft and pending jobs from the `fetchJobs` action, a find is required to search. 

Also, if we get an empty response from `fetchJobs` actions and find is ran, the `catch` block is executed.

File: src/store/modules/job/actions.ts [hunk 2]
```diff
@@ -464,17 +464,18 @@ const actions: ActionTree<JobState, RootState> = {
...
           inputFields: {
             'systemJobEnumId': payload.systemJobEnumId,
             'systemJobEnumId_op': 'equals'
           }
         })
-        const fetchJobsResponse = jobs.find((job: any) => job?.jobId === payload.jobId);
-        if(fetchJobsResponse.status === 200 && !hasError(fetchJobsResponse) && fetchJobsResponse.data?.docs.length) {
+        // As we are getting both draft and pending jobs in response, we are using find
+        const jobResponse = fetchJobResponses.find((response: any) => response.data.docs[0].jobId === payload.jobId);
+        if(jobResponse.status === 200 && !hasError(jobResponse)) {
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:475]:
```suggestion
        const jobResponse = fetchJobResponses.find((response: any) => fetchJobsResponse.status === 200 && !hasError(fetchJobsResponse) && fetchJobsResponse.data?.docs.length && response.data.docs[0].jobId === payload.jobId);
        if(jobResponse) {
```

File: src/store/modules/job/actions.ts [hunk 3]
```diff
@@ -464,17 +464,18 @@ const actions: ActionTree<JobState, RootState> = {
...
           inputFields: {
             'systemJobEnumId': payload.systemJobEnumId,
             'systemJobEnumId_op': 'equals'
           }
         })
-        const fetchJobsResponse = jobs.find((job: any) => job?.jobId === payload.jobId);
-        if(fetchJobsResponse.status === 200 && !hasError(fetchJobsResponse) && fetchJobsResponse.data?.docs.length) {
+        // As we are getting both draft and pending jobs in response, we are using find
+        const jobResponse = fetchJobResponses.find((response: any) => fetchJobsResponse.status === 200 && !hasError(fetchJobsResponse) && fetchJobsResponse.data?.docs.length && response.data.docs[0].jobId === payload.jobId);
+        if(jobResponse) {
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:475]:
```suggestion
        const jobResponse = fetchJobResponses.find((response: any) => response.status === 200 && !hasError(response) && response.data?.docs.length && response.data.docs[0].jobId === payload.jobId);
        if(jobResponse) {
```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**